src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java

```diff
@@ -294,6 +317,44 @@ public FileResource(FileService fileService, ResourceLoaderService resourceLoade
         return buildFileResponse(Paths.get(FilePathService.getLectureAttachmentFilePath(), String.valueOf(optionalLecture.get().getId())).toString(), filename);
     }
 
+    /**
+     * GET /files/attachments/lecture/{lectureId}/merge-pdf : Get the lecture units
+     * PDF attachments merged
+     *
+     * @param lectureId            ID of the lecture, the lecture units belongs to
+     * @param temporaryAccessToken The access token is required to authenticate the
+     *                             user that accesses it
+     * @return The merged PDF file, 403 if the logged in user is not allowed to
+     *         access it, or 404 if the files to be merged do not exist
+     */
+    @GetMapping("files/attachments/lecture/{lectureId}/merge-pdf")
+    @PreAuthorize("permitAll()")
+    public ResponseEntity<byte[]> getLecturePdfAttachmentsMerged(@PathVariable Long lectureId, @RequestParam("access_token") String temporaryAccessToken) {
+        log.debug("REST request to get merged pdf files for a lecture with id : {}", lectureId);
```

Comment:
I've used the /files/attachments/access-token/{filename:.+} endpoint to create the token and the validateTemporaryAccessToken method as they are used to access the other attachment files but I see that actually there is no check whether the user has access to this course/attachment. 
You are right that if the user manages to change the lecture id in the URL within 30 seconds which is the validation time of the token and enters valid id he can access files for other lectures.

Reply:
Stupid question: Why is it necessary to use a token instead of checking if the user is allowed to view course contents directly?

Reply:
I implemented it in the same logic as it was implemented for downloading file attachments but I think the reason for that is because we want to open the file in a new window where a preview is available